blockonomics-woocommerce.php

```diff
@@ -501,10 +501,16 @@ function blockonomics_activation_hook() {
 }
 
 // Since WP 3.1 the activation function registered with register_activation_hook() is not called when a plugin is updated.
+// Adding timestamp column in db to handle underpayments
 function blockonomics_update_db_check() {
     global $wpdb;
     global $blockonomics_db_version;
 
+    $timestamp_column = $wpdb->get_results(  "SELECT COLUMN_NAME FROM INFORMATION_SCHEMA.COLUMNS
+    WHERE table_name = 'wp_blockonomics_orders' AND column_name = 'timestamp'"  );
+    if(empty($timestamp_column)){
+        $wpdb->query("ALTER TABLE wp_blockonomics_orders ADD timestamp INT DEFAULT NULL");
+    }
     $installed_ver = get_site_option( 'blockonomics_db_version' );
     if ( $installed_ver != $blockonomics_db_version ) {
```

Comment:
Upgrade should happen here only trigger the migration code related to upgrade. I am not sure why we are running blockonomics_create_table() for everything. @DarrenWestwood  can you elaborate on this

Another thing to think about is that existing orders will have timestamp NULL. We have to think through implications of this if any. If underpayment doesn't work on existing orders its ok, but ongoing callbacks to existing orders should not get effected

Reply:
> Upgrade should happen here only trigger the migration code related to upgrade. I am not sure why we are running blockonomics_create_table() for everything.

Yes blockonomics_create_table should only be run if there is no $installed_ver. I have opened a separate issue to deal with this:  https://github.com/blockonomics/woocommerce-plugin/issues/296

> Another thing to think about is that existing orders will have timestamp NULL

Adding timestamp may not be necessary as we should be able to simply check for pending address: https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/master/modules/gateways/blockonomics/blockonomics.php#L426-L436 (There can only be one pending address at a time for each crypto, therefore it is not necessary to order addresses by timestamp to locate the pending address)